mod/settings.php

```diff
@@ -576,41 +562,17 @@ function settings_post(App $a)
 		}
 	}
 
+	$fields = ['username' => $username, 'email' => $email, 'timezone' => $timezone,
+		'allow_cid' => $str_contact_allow, 'allow_gid' => $str_group_allow, 'deny_cid' => $str_contact_deny, 'deny_gid' => $str_group_deny,
+		'notify-flags' => $notify, 'page-flags' => $notify, 'account-type' => $account_type, 'default-location' => $defloc,
+		'allow_location' => $allow_location, 'maxreq' => $maxreq, 'expire' => $expire, 'def_gid' => $def_gid, 'blockwall' => $blockwall,
+		'hidewall' => $hide_wall, 'blocktags' => $blocktags, 'unkmail' => $unkmail, 'cntunkmail' => $cntunkmail, 'language' => $language];
 
-	$r = q("UPDATE `user` SET `username` = '%s', `email` = '%s',
-				`openid` = '%s', `timezone` = '%s',
-				`allow_cid` = '%s', `allow_gid` = '%s', `deny_cid` = '%s', `deny_gid` = '%s',
-				`notify-flags` = %d, `page-flags` = %d, `account-type` = %d, `default-location` = '%s',
-				`allow_location` = %d, `maxreq` = %d, `expire` = %d, `openidserver` = '%s',
-				`def_gid` = %d, `blockwall` = %d, `hidewall` = %d, `blocktags` = %d,
-				`unkmail` = %d, `cntunkmail` = %d, `language` = '%s'
-			WHERE `uid` = %d",
-			DBA::escape($username),
-			DBA::escape($email),
-			DBA::escape($openid),
-			DBA::escape($timezone),
-			DBA::escape($str_contact_allow),
-			DBA::escape($str_group_allow),
-			DBA::escape($str_contact_deny),
-			DBA::escape($str_group_deny),
-			intval($notify),
-			intval($page_flags),
-			intval($account_type),
-			DBA::escape($defloc),
-			intval($allow_location),
-			intval($maxreq),
-			intval($expire),
-			DBA::escape($openidserver),
-			intval($def_gid),
-			intval($blockwall),
-			intval($hidewall),
-			intval($blocktags),
-			intval($unkmail),
-			intval($cntunkmail),
-			DBA::escape($language),
-			intval(local_user())
-	);
-	if (DBA::isResult($r)) {
+	if ($delete_openid) {
+		$fields['openid'] = '';
+		$fields['openidserver'] = '';
+	}
```

Comment:
Code standards: Please add a new line after closing brace if the next line isn't another closing brace either.